pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -232,6 +254,17 @@ func (am *alertmanager) ApplyConfig(ctx context.Context, dbCfg *ngmodels.AlertCo
 
 	var outerErr error
 	am.Base.WithLock(func() {
+		if am.withAutogen {
+			err := AddAutogenConfig(ctx, am.logger, am.Store, am.orgID, &cfg.AlertmanagerConfig, true)
+			if err != nil {
+				outerErr = err
+				return
+			}
+		}
+		// Note: Adding the autogen config here causes alert_configuration_history to update last_applied more often.
+		// Since we will now update last_applied when autogen changes even if the user-created config remains the same.
+		// To fix this however, the local alertmanager needs to be able to tell the difference between user-created and
+		// autogen config, which may introduce cross-cutting complexity.
```

Comment:
I think that's how it supposed to work. The database contains "user-defined" routes and therefore the history and hash  should contain only that data.

Reply:
Previously last_applied would only update if the db config changed. Now last_applied can update even when the db config stays unchanged (if the autogen part changes).

This is not necessarily an issue, in fact I believe it might be more correct than the alternative. The main oddity is that since we don't store anything autogen related (config or hash) in the db, last_applied can update with no discernible change in db state.